cmd/root.go

```diff
@@ -41,12 +40,34 @@ func Execute() {
 }
 
 func init() {
-	RootCmd.PersistentFlags().StringVarP(&rootConfig.kubeConfig, "kubeconfig", "c", "", "config file (default is $HOME/.kube/config")
+	cobra.OnInitialize(processFlags)
+	RootCmd.PersistentFlags().BoolVarP(&rootConfig.localMode, "local", "l", false, "[DEPRECATED] Local mode, uses $HOME/.kube/config as configuration")
+	RootCmd.Flags().MarkHidden("local")
+	RootCmd.PersistentFlags().StringVarP(&rootConfig.kubeConfig, "kubeconfig", "c", "", "Specify local config file (default is $HOME/.kube/config")
 	RootCmd.PersistentFlags().StringVarP(&rootConfig.verbose, "verbose", "v", "INFO", "Set the debug level")
-	RootCmd.PersistentFlags().BoolVarP(&rootConfig.localMode, "local", "l", false, "Local mode, uses ~/.kube/config as configuration")
 	RootCmd.PersistentFlags().BoolVarP(&rootConfig.json, "json", "j", false, "Enable json logging")
 	RootCmd.PersistentFlags().BoolVarP(&rootConfig.allPods, "allPods", "a", false, "Audit againsts pods in all the phases (default Running Phase)")
 	RootCmd.PersistentFlags().StringVarP(&rootConfig.namespace, "namespace", "n", apiv1.NamespaceAll, "Specify the namespace scope to audit")
 	RootCmd.PersistentFlags().StringVarP(&rootConfig.manifest, "manifest", "f", "", "yaml configuration to audit")
 	RootCmd.PersistentFlags().StringVarP(&rootConfig.dropCapConfig, "dropCapConfig", "d", "", "yaml configuration to audit")
 }
+
+func processFlags() {
+	if rootConfig.verbose == "DEBUG" {
+		log.SetLevel(log.DebugLevel)
+		log.AddHook(NewDebugHook())
+	}
+	if rootConfig.json {
+		log.SetFormatter(&log.JSONFormatter{})
+	}
+
+	if rootConfig.localMode == true {
```

Comment:
That's how kubeaudit handled `-l` prior to my changes. Here is a sample from the current master branch:
```
./kubeaudit -l version
INFO[0000] Kubeaudit                                     Version=0.1.0
INFO[0000] Kubernetes server version                     Major=1 Minor=10+ Platform=linux/amd64
INFO[0000] Kubernetes client version                     Major= Minor= Platform=darwin/amd64

./kubeaudit -l -c /notarealfile version
INFO[0000] Kubeaudit                                     Version=0.1.0
INFO[0000] Kubernetes server version                     Major=1 Minor=10+ Platform=linux/amd64
INFO[0000] Kubernetes client version                     Major= Minor= Platform=darwin/amd64
```

Since I only kept -l in for backwards compatibility, I thought it best to maintain the existing behaviour. However, I agree that it's not the behaviour one would expect and that it would be improved by having `-c` override `-l`.

It's a trivial fix so just let me know if you still think I should change it.

Reply:
I think if we are already touching the code it would be cool to fix this bug :) But yeah I can see where you are coming from and honestly I didn't know it behaved like that.

Reply:
Fixed in https://github.com/Shopify/kubeaudit/pull/113/commits/4f678c4e97f5e651298dd5f0cbd6192fe41c9bf0. This is the new behaviour:
```
./kubeaudit -l -c /notarealfile version
WARN[0000] -l/-local is deprecated! kubeaudit will default to local mode if it's not running in a cluster.
INFO[0000] Kubeaudit                                     Version=0.1.0
ERRO[0000] Unable to load kubeconfig. Could not open file /notarealfile.
WARN[0000] Could not get kubernetes server version.
```